### PR TITLE
#181: nDPI: Commit nDPI Version 02.29.2024.1

### PR DESCRIPTION
**Pull request by akumartrapp**
Please sign (check) the below before submitting the Pull Request:

- [ ] I have signed the ntop Contributor License Agreement at https://github.com/ntop/legal/blob/main/individual-contributor-licence-agreement.md
- [ ] I have read the contributing guide lines at https://github.com/ntop/nDPI/blob/dev/CONTRIBUTING.md
- [ ] I have updated the documentation (in doc/) to reflect the changes made (if applicable)

Link to the related [issue](https://github.com/ntop/nDPI/issues):

Describe changes:

### CODE REVIEW

**Comment by IvanNardi** [comment]
@akumartrapp, are you sure you pushed this code to the right repository? It seems this PR is meant for a your personal fork...

File: example/ReadJson.c [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+{
+    // Get the folder path of the currently running program
+    char* end;
+    size_t length;
+
+#ifdef _WIN32
+    GetModuleFileName(NULL, buffer, size);
+    end = strrchr(buffer, '\\');
+#else
+    if (readlink("/proc/self/exe", buffer, size) == -1) {
```
Comment (utoni) [inline example/ReadJson.c:31]: won't work on mac

File: example/ReadJson.c [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+        }
+        else if (type == json_type_array)
+        {
+            int arrayLength = json_object_array_length(val);
+
+            for (int i = 0; i < arrayLength; ++i)
+            {
+                *vectorSize += 1;
+                *paramsVector = realloc(*paramsVector, (*vectorSize) * sizeof(struct SkipParameters));
+                (*paramsVector)[*vectorSize - 1].sourceIP = _strdup("NOT_SET");
```
Comment (utoni) [inline example/ReadJson.c:78]: Why even set those keys?

File: example/ndpiReader.c [hunk 3]
```diff
@@ -559,7 +582,6 @@ static void help(u_int long_help) {
          "  -c <path>                 | Load custom categories from the specified file\n"
          "  -C <path>                 | Write output in CSV format on the specified file\n"
          "  -r <path>                 | Load risky domain file\n"
-         "  -R                        | Print detected realtime protocols\n"
```
Comment (utoni) [inline example/ndpiReader.c:562]: ???

File: example/ndpiReader.c [hunk 4]
```diff
@@ -1017,7 +1200,7 @@ static void parseOptions(int argc, char **argv) {
 
     case 'i':
     case '3':
-      _pcap_file[0] = optarg;
+     // _pcap_file[0] = optarg;  // Ashwani: This is commented out as we take pcap/pcapng files from the input folder which is set using option '-R'
```
Comment (utoni) [inline example/ndpiReader.c:1215]: ???

File: example/ndpiReader.c [hunk 5]
```diff
@@ -1083,8 +1266,8 @@ static void parseOptions(int argc, char **argv) {
       break;
 
     case 'R':
-      enable_realtime_output =1;
-      break;
+        pCapFilesFolderLocationPath = optarg; // Ashwani: This is folder path of pcap/pcapng files
```
Comment (utoni) [inline example/ndpiReader.c:1281]: What happened to the old option?

**Review by utoni — CHANGES_REQUESTED** [review]
I do not understand what the idea of the PR was.
Did you want to replace the old JSON serialization or process pcap files from a directory recursively or both or something else?

**Comment by IvanNardi** [comment]
@akumartrapp, any updates?

**Comment by utoni** [comment]
Closing for inactivity. Please re-open if you still want to get this patch into `dev`.

**Pull request closed without merging**